azure/pf-azure-context/classes/pf-azure-context.ps1

```diff
@@ -1,32 +1,32 @@
 Class PfAzureContext : PfCloudContext{
     [string]$SubscriptionName
     [string]$ResourceGroupName
-    [string]$Region
-    [string]$TeamAbbreviation
+    [AzRegion]$AzRegion
+    [string]$Label
     [string]$CompanyAbbreviation
     [string]$GroupAbbreviation
     [string]$EnvironmentLetter
 
-    PfAzureContext([string]$SubscriptionName,[string]$ResourceGroupName,[string]$Region){
+    PfAzureContext([string]$SubscriptionName,[string]$ResourceGroupName,[AzRegion]$AzRegion){
```

Comment:
I thinking about concept of a defaults config state that we set up for people that they just populate and put somewhere in the folder hiearchy for running PF, and then some of these things like Region can be defaulted there, so don't need to go on every instantiation of azure context.  Basically, the environment config, which could also be loaded from state instead of all file by putting in config for loading state from external source instead of file

This then also answers TODO elsewhere of how to override config from deployment.

Reply:
Also possibly applies to some of the other context constructor options, like co + group abbrev, label.  Sub and RG are always going to be supplied somewhere in the deploy though (on context or resources)

Reply:
So are you saying that we need to have a function then that would be "Import-PfEnvironmentConfig" and then take in a path to a json config file that we would then store in a variable?  And then that config function need to be the first thing instantiated within a deployment script?

Reply:
Could be something like that, or just convention based file in location that we load, and that is populated by defaults.  So we can support all the complex scenarios but keep even more things out of needing to be set/done if someone just wants to do the 'hello world'.

Reply:
like maybe a pf-config.json that sits next to deploy script, and when a context 'comes to life' it goes and looks for this file.  could even climb dir tree looking for it.  this could be thing that would config how to do state management also.